### PR TITLE
40 add basic system tests

### PR DESCRIPTION
**Pull request by lfse-slafleur**

### CODE REVIEW

**Comment by lfse-slafleur** [comment]
@lfse-slafleur  Do not forget to merge main into this branch and add the resulting ESDL's from optimizer & simulator.

**Review by MarkTNO — APPROVED** [review]
Very cool to have these system test available!

Some questions:
- For the tests, should the output esdl file path be moved to the `# Arrange` step?
- Should we add the `create_venv.sh` and update/install dependencies scripts?
- We probably don't want to add linting/type checking for the one testing script?

File: scripts/test_system.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+. scripts/_select_docker_compose.sh
+
+export COMPOSE_PROJECT_NAME=system_tests
```
Comment (MarkTNO) [inline scripts/test_system.sh:4]: Maybe change to `omotes_system_tests`?

**Comment by lfse-slafleur** [comment]
> Very cool to have these system test available!
> 
> Some questions:
> 
>     * For the tests, should the output esdl file path be moved to the `# Arrange` step?
I would prefer to keep them in the # Assert section for readability.
> 
>     * There is a lot of print output during the tests, mainly due to the worker output. This makes it a bit harder to solve issues. Could this be reduced?

Good question. I had a look at some of the alternatives but I haven't found any options that I think would work better. We could move the system tests themselves outside of the docker container and let the test script manage the environment through docker-compose. As it would then be a subprocess, it can listen for stdout / stderr and attach each of them to each of the tests. However, this may prove complex in handling. Something like this may help: https://github.com/avast/pytest-docker . However, if a previous test screws something up but the test succeeds, the issue will be silenced as pytest will only show stdout/stderr for the test that fails. A clean environment per test (delete all containers and volumes after each test) is too heavy as setting up takes quite some time.

Another option would be to write the logs to file, however, it would write ALL logs to file. I haven't found an alternative where we can write the logs for some containers to a file somewhere on disk, and some not. Regardless, if there is then an issue, you will need to look through the files anyway.

While the output is a bit hard to filter through, the current setup is simple and ALL logs are shown which is helpful in a debug scenario. However, it is hard to relate where each test currently starts and which section of the logs belong to which tests. Maybe we can clarify this by printing when a test starts and ends. How to circumvent this in pytest is unknown to me but perhaps it is possible.
> 
>     * Should we add the `create_venv.sh` and update/install dependencies scripts?
You mean to the system_tests folder? We definitely could for running the test manually!
> 
>     * We probably don't want to add linting/type checking for the one testing script?
Honestly, makes sense to add it as this test suite is expected to grow.

File: system_tests/src/test_workflows_steps.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+            WorkflowType("grow_optimizer_default", ""),
+            WorkflowType("grow_simulator", ""),
+            WorkflowType("grow_optimizer_no_heat_losses", ""),
+            WorkflowType("grow_optimizer_with_pressure", ""),
+            WorkflowType("simulator", ""),
+        ]
+    )
+
+
+def omotes_job_result_handler() -> OmotesJobHandler:
```
[the remark this answers is not in the document]
Comment (lfse-slafleur) [reply system_tests/src/test_workflows_steps.py]: Removed

File: scripts/start.sh [hunk 3]
```diff
@@ -2,5 +2,5 @@
 
 . scripts/_select_docker_compose.sh
 
-$DOCKER_COMPOSE --profile=manual_dev --profile=manual_setup down
-$DOCKER_COMPOSE --profile mapeditor-dispatcher up -d
+$DOCKER_COMPOSE --profile=manual_dev down orchestrator_postgres_db_dev
+$DOCKER_COMPOSE up -d
```
[the remark this answers is not in the document]
Comment (lfse-slafleur) [reply scripts/start.sh]: Done

File: scripts/test_system.sh (hunk 1 above)
Comment (lfse-slafleur) [reply scripts/test_system.sh]: Done

**Comment by lfse-slafleur** [comment]
To capture the discussed work:
- https://github.com/Project-OMOTES/computation-engine/issues/48
- https://github.com/Project-OMOTES/computation-engine/issues/47

**Pull request merged**